builder/simplify.cc

```diff
@@ -391,25 +401,44 @@ class simplifier : public node_mutator {
       set_result(let_stmt::make(std::move(lets), std::move(body)));
     }
   }
+  
+  void merge_references(symbol_map<reference_info> add) {
```

Comment:
This seems more specifically to be a function that merges references from a loop. I'd rename it to `merge_loop_references` or something.